near-sdk/src/lib.rs

```diff
@@ -99,10 +101,321 @@
+/// Or use `#[near(serializers=[json])]` to make it serializable with Json. By default, borsh is used.
+/// You can also specify both and none. BorshSchema or JsonSchema are always generated.
+///
+/// If you want the struct to be a contract state, you can pass in the contract_state argument.
+///
+/// ## Example
+/// ```rust
+/// use near_sdk::near;
+///
+/// #[near(contract_state)]
+/// pub struct Contract {
+///    data: i8,
+/// }
+///
+/// #[near]
+/// impl Contract {
+///     pub fn some_function(&self) {}
+/// }
+/// ```
+/// As well, the macro supports arguments like `event_json` and `contract_metadata`.
+///
+/// # Events Standard:
+///
+/// By passing `event_json` as an argument `near_bindgen` will generate the relevant code to format events
+/// according to NEP-297
+///
+/// For parameter serialization, this macro will generate a wrapper struct to include the NEP-297 standard fields `standard` and `version
+/// as well as include serialization reformatting to include the `event` and `data` fields automatically.
+/// The `standard` and `version` values must be included in the enum and variant declaration (see example below).
+/// By default this will be JSON deserialized with `serde`
+///
+///
+/// ## Examples
+///
+/// ```rust
+/// use near_sdk::{near, AccountId};
+///
+/// # #[near(contract_state)]
+/// # pub struct Contract {
+/// #    data: i8,
+/// # }
+///
+///
+/// #[near(event_json(standard = "nepXXX"))]
+/// pub enum MyEvents {
+///    #[event_version("1.0.0")]
+///    Swap { token_in: AccountId, token_out: AccountId, amount_in: u128, amount_out: u128 },
+///
+///    #[event_version("2.0.0")]
+///    StringEvent(String),
+///
+///    #[event_version("3.0.0")]
+///    EmptyEvent
+/// }
+///
+/// #[near]
+/// impl Contract {
+///     pub fn some_function(&self) {
+///         MyEvents::StringEvent(
+///             String::from("some_string")
+///         ).emit();
+///     }
+///
+/// }
+/// ```
+///
+/// # Contract Source Metadata Standard:
+///
+/// By using `contract_metadata` as an argument `near` will populate the contract metadata
+/// according to [`NEP-330`](<https://github.com/near/NEPs/blob/master/neps/nep-0330.md>) standard. This still applies even when `#[near]` is used without
+/// any arguments.
+///
+/// All fields(version, link, standard) are optional and will be populated with defaults from the Cargo.toml file if not specified.
+///
+/// The `contract_source_metadata()` view function will be added and can be used to retrieve the source metadata.
+/// Also, the source metadata will be stored as a constant, `CONTRACT_SOURCE_METADATA`, in the contract code.
+///
+/// ## Examples
+/// ```rust
+/// use near_sdk::near;
+///
+/// #[near(contract_metadata(
+///     version = "39f2d2646f2f60e18ab53337501370dc02a5661c",
+///     link = "https://github.com/near-examples/nft-tutorial",
+///     standard(standard = "nep330", version = "1.1.0"),
```

Comment:
the line with explicit  330 should be removed in example, i'll add the sc link of where it's populated automatically

Reply:
https://github.com/near/near-sdk-rs/blob/master/near-sdk-macros/src/core_impl/contract_metadata/mod.rs?plain=1#L75